### PR TITLE
build: install linker files for installed targets

### PR DESCRIPTION
**Pull request by compnerd**
Ensure that we install any linker files for the modules which are required on certain platforms.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

File: Sources/FoundationInternationalization/CMakeLists.txt
Comment (jmschonfeld) [inline Sources/FoundationInternationalization/CMakeLists.txt:57]: nit: should we just add this to `_swift_foundation_install_target` so that this is automatically applied for any targets we might install in the future?

Comment (compnerd) [reply Sources/FoundationInternationalization/CMakeLists.txt:57]: Sure, let me sink it into that.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by jmschonfeld** [comment]
Thanks, this seems fine to me! Before merging could we kick off a toolchain build or full swift test with this change just to get some good coverage over the CMake changes? We don't have that setup on this repo yet so it'll need to be kicked off via a PR to the swiftlang/swift repo (or the swift-corelibs-foundation repo) to double check this doesn't cause any issues in the toolchain build on linux+windows

**Comment by compnerd** [comment]
Sure! Kicked that off at https://github.com/swiftlang/swift/pull/65907. I'll follow up with the installer changes later, I want to get this in to unblock CI

**Review by jmschonfeld — APPROVED** [review]
Awesome thanks! Approved and good to merge after that build finishes successfully

**Comment by compnerd** [comment]
https://ci-external.swift.org/job/swift-PR-build-toolchain-windows/5374/ was the run.

**Pull request merged**
